xdsl/transforms/stencil_to_csl_stencil.py

```diff
@@ -0,0 +1,173 @@
+from collections.abc import Sequence
+
+from attr import dataclass
+
+from xdsl.context import MLContext
+from xdsl.dialects import memref, stencil, tensor
+from xdsl.dialects.builtin import ModuleOp, TensorType
+from xdsl.dialects.csl import csl_stencil
+from xdsl.dialects.experimental import dmp
+from xdsl.ir import Attribute, OpResult
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    GreedyRewritePatternApplier,
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.utils.hints import isa
+
+
+@dataclass(frozen=True)
+class AccessOpFromPrefetchPattern(RewritePattern):
+    """
+    Rebuilds stencil.access by csl_stencil.access which operates on prefetched accesses.
+
+    stencil.access operates on stencil.temp types found at arg_index
+    csl_stencil.access operates on memref< num_neighbors x tensor< buf_size x data_type >> found at last arg index
+
+    Note: This is intended to be called in a nested pattern rewriter, such that the above precondition is met.
+    """
+
+    arg_index: int
```

Comment:
`arg_index` is not actually used in this pattern

Reply:
Fixing this.